app/src/main/java/org/p2p/wallet/send/ui/dialogs/SendConfirmBottomSheet.kt

```diff
@@ -82,6 +93,37 @@ class SendConfirmBottomSheet(
         }
     }
 
+    private fun initPinView() = with(binding) {
+        pinView.apply {
+            setFingerprintVisible(true)
+            onPinCompleted = ::checkPinCode
+            onBiometricClicked = {
+                isVisible = false
+                confirmBiometrics()
+            }
+        }
+    }
+
+    private fun confirm() {
+        onConfirmed.invoke()
+        dismissAllowingStateLoss()
+    }
+
+    private fun checkPinCode(pinCode: String) {
+        CoroutineScope(dispatchers.io).launch {
```

Comment:
Redundant use of coroutines, and we should avoid launching not lifecycle aware  coroutines